rules/throw-new-error.js

```diff
@@ -23,6 +23,18 @@ const create = context => ({
 			return;
 		}
 
+		// https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2654 (Effect library)
+		if (
+			callee.type === 'MemberExpression'
+			&& !callee.computed
+			&& callee.object.type === 'Identifier'
+			&& callee.object.name === 'Data'
+			&& callee.property.type === 'Identifier'
+			&& callee.property.name === 'TaggedError'
+		) {
```

Comment:
Let's use `isMemberExpression` or  `isNodeMatches`.

Reply:
Done in [`70c9ddc` (#2740)](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2740/commits/70c9ddc3af4a0b6ee325285dfda299c26b4892e5)